app/views/posts/index.html.erb

```diff
@@ -1,2 +1,44 @@
-<h1>All Posts</h1>
-<!-- Placeholder for listing all posts by a user -->
+<h2>All Posts</h2>
+<main>
+  <div class="user-container">
+    <div class="profile-pic">
+      <img src="<%= @user.photo %>" alt="profile-picture">
+    </div>
+    <div class="user-info">
+      <h3><%= @user.name %></h3>
+      <p class="posts-counter">Number of posts: <%= @user.posts_counter || 0 %></p>
+    </div>
+  </div>
+
+  <article>
+    <% @posts.each do |post| %>
+      <%= link_to user_post_path(@user, post) do %>
+        <div class="post-container">
+          <h5 class="post-title"><%= post.title %></h5>
+          <% if post.text.present? %>
+            <p><%= post.text[0..100] %>...</p>
+          <% else %>
+            <p>No text available...</p>
+          <% end %>
+        </div>
+        <div class="comment-likes-div">
+          <p>Comment: <%= post.comments_counter || 0 %>, Likes: <%= post.likes_counter || 0 %></p>
+        </div>
+        <ul class="post-comments-div">
+          <% if post.five_most_recent_comments.any? %>
+            <% post.five_most_recent_comments.each do |comment| %>
+              <li>
+                <p>
+                  <%= User.find_by(id: comment.author_id).name if comment.author_id %>:
+                  <%= comment.text[0..60] if comment.text.present? %>...
+                </p>
+              </li>
+            <% end %>
+          <% else %>
+            <li>No comments available...</li>
+          <% end %>
+        </ul>
+      <% end %>
+    <% end %>
+  </article>
+</main>
```

Comment:
| @WilhelmK109 Post Index Page | Microverse Post Index Page |
|-|-|
| ![image](https://github.com/WilhelmK109/BlogApp/assets/93039359/9f9ea9d0-c794-47e9-91a2-22dff3b775ea) | ![image](https://github.com/WilhelmK109/BlogApp/assets/93039359/ed74a507-e1f9-4044-9d16-b35eb3f915c1) |

- [ ] Great work so far but here we need minor changes 😊 Kindly expand your data on the whole screen as you can see in the given design 👀 that the Name is on the left side and Posts Counter is on the right side 🤔 to organize the user information well 👍 Kindly make this change 📝
- [ ] It would be nice if you made the Pagination button in your design as you can see in the given design 👍 to make your app look professional 😊